tools/bug-report/pkg/content/content.go

```diff
@@ -0,0 +1,157 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package content
+
+import (
+	"fmt"
+	"strings"
+
+	"istio.io/istio/tools/bug-report/pkg/kubectlcmd"
+	"istio.io/pkg/log"
+)
+
+const (
+	coredumpDir = "/var/lib/istio"
+)
+
+var (
+	istiodURLs = []string{
+		"debug/configz",
+		"debug/endpointz",
+		"debug/adsz",
+		"debug/authenticationz",
+		"metrics",
+	}
+)
+
+// Params contains parameters for running a kubectl fetch command.
+type Params struct {
+	DryRun    bool
+	Verbose   bool
+	Namespace string
+	Pod       string
+	Container string
+}
+
+// GetK8sResources returns all k8s cluster resources.
+func GetK8sResources(params *Params) (string, error) {
```

Comment:
I am not sure I understand this. First we use client-go to fetch all of the resources in the cluster. Then later we use kubectl? Why do we mix them? And seems like we fetch them all.

Also if `all` is fetched why do we get these other things?

Reply:
client-go gives a structured output so we can build up a cluster resource tree. can't use kubectl for that.
we could in principle use client-go to grab the outputs, but a) some of the commands are very convoluted to express in go-client so the fetch part would be quite verbose and harder to read and b) it's better to keep this part in a format that's easily extensible for someone who wants to grab another artifact without coming up to speed on all the code.

Reply:
regarding "all", it't not really all, it doesn't include the other items.

Reply:
If we really need the dynamicness of something like `kubectl get all,jobs` rather than directly client-go interaction we can also use something like https://pkg.go.dev/k8s.io/cli-runtime/pkg/genericclioptions?tab=doc. But really the dynamic client is probably more appropriate